bitcoin/script/miniscript.h

```diff
@@ -293,6 +294,41 @@ struct StackSize {
     StackSize(MaxInt<uint32_t> in_sat, MaxInt<uint32_t> in_dsat) : sat(in_sat), dsat(in_dsat) {};
 };
 
+struct TimeLockInfo {
```

Comment:
Any reason why this is a separate structure, and not integrated with the other type properties?

Reply:
When we calculate something as a type property, we get the benefit of caching intermediate nodes in the Miniscript struct. I didn't see any use in saving this information for intermediate miniscripts nodes, I did not make them type properties. 

We decided
1) The Miniscripts that contain a possible spend of timelock and heightlock are still well-typed valid miniscripts. 
2) For the policy compiler, we refuse to accept policies that can contain such a spend path
3) We don't do any semantic analysis on such Miniscripts because that can result in incorrect results. 

If you think I should create five more types in type-properties, I can change that too. I don't have any strong preference.